File: 0x14-javascript-web_scraping/3-starwars_title.go
package main

import (
	"encoding/json"
	"fmt"
	"io"
	"log"
	"net/http"
	"os"
)

type Movie struct {
	Title     string `json:"title"`
	EpisodeID int    `json:"episode_id"`
}

func main() {
	if len(os.Args) != 2 {
		fmt.Println("Usage: <app> <movieID>")
		return
	}

	movieID := os.Args[1]
	url := "https://swapi-api.alx-tools.com/api/films/"

	response, err := http.Get(url + movieID)
	if err != nil {
		log.Fatalln(err)
	}
	defer response.Body.Close()

	body, err := io.ReadAll(response.Body)
	if err != nil {
		log.Fatalf("Failed to read response body: %s", err)
	}

	var movie Movie
	err = json.Unmarshal(body, &movie)
	if err != nil {
		log.Fatalf("Failed to unmarshal movie details: %s", err)
	}

	fmt.Println(movie.Title)

}

File: 0x14-javascript-web_scraping/3-starwars_title.js
#!/usr/bin/node
const request = require('request');

const movieID = process.argv[2];
const apiURL = `https://swapi-api.alx-tools.com/api/films/${movieID}`;

request.get(apiURL, (err, response, body) => {
  if (err) {
    console.error(err);
  } else if (response.statusCode === 200) {
    const movieData = JSON.parse(body);
    console.log(movieData.title);
  } else {
    console.log(`Error code: ${response.statusCode}`);
  }
});
